### PR TITLE
project: Upstream merge

### PR DESCRIPTION
**Pull request by zeruth**
Still needs varbits for SOTE, and In Search of Knowledge

### CODE REVIEW

File: http-api/src/main/java/net/runelite/http/api/ws/messages/party/PartyChatMessage.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.http.api.ws.messages.party;
+
+import lombok.Value;
+
+@Value
+public class PartyChatMessage extends PartyMemberMessage
```
Comment (Owain94) [inline http-api/src/main/java/net/runelite/http/api/ws/messages/party/PartyChatMessage.java]:
```suggestion
public class PartyChatMessage extends PartyMemberMessage implements Event
```

File: http-api/src/main/java/net/runelite/http/api/ws/messages/party/PartyChatMessage.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.http.api.ws.messages.party;
+
+import lombok.Value;
+
```
Comment (Owain94) [inline http-api/src/main/java/net/runelite/http/api/ws/messages/party/PartyChatMessage.java]:
```suggestion
import net.runelite.api.events.Event;
```

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java [hunk 3]
```diff
@@ -270,6 +274,7 @@ private void addSubscriptions()
 		eventBus.subscribe(ClientTick.class, this, this::onClientTick);
 		eventBus.subscribe(MenuEntryAdded.class, this, this::onMenuEntryAdded);
 		eventBus.subscribe(FocusChanged.class, this, this::onFocusChanged);
+				eventBus.subscribe(MenuOptionClicked.class, this, this::onMenuOptionClicked);
```
Comment (Owain94) [inline runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java]: Tabs

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java [hunk 4]
```diff
@@ -199,6 +202,28 @@
...
 	private boolean raidStarted;
+
+	@Inject
+	private PartyService party;
+
+	@Inject
+	private WSClient ws;
+
+	@Getter
+	private final ArrayList<String> roomWhitelist = new ArrayList<>();
```
Comment (Owain94) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java]:
```suggestion
	private final List<String> roomWhitelist = new ArrayList<>();
```

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java [hunk 5]
```diff
@@ -199,6 +202,28 @@
...
+
+	@Getter
+	private final ArrayList<String> roomBlacklist = new ArrayList<>();
```
Comment (Owain94) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java]:
```suggestion
	private final List<String> roomBlacklist = new ArrayList<>();
```

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java [hunk 6]
```diff
@@ -199,6 +202,28 @@
...
+
+	@Getter
+	private final ArrayList<String> rotationWhitelist = new ArrayList<>();
```
Comment (Owain94) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java]:
```suggestion
	private final List<String> rotationWhitelist = new ArrayList<>();
```

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java [hunk 7]
```diff
@@ -199,6 +202,28 @@
...
+
+	@Getter
+	private final ArrayList<String> layoutWhitelist = new ArrayList<>();
```
Comment (Owain94) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPlugin.java]:
```suggestion
	private final List<String> layoutWhitelist = new ArrayList<>();
```

**Comment by Kyleeld** [comment]
https://github.com/runelite/runelite/commit/bd7a6ed7ef2cbfe2ceb6c50c837c74c064d54bd4

Mournings end needs updating

**Pull request merged**